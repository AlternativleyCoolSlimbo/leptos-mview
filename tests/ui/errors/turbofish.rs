use leptos::*;
use leptos_mview::view;

fn main() {
    use core::marker::PhantomData;
    // copied from https://github.com/leptos-rs/leptos/pull/1636
    #[component]
    pub fn GenericComponent<S>(ty: PhantomData<S>) -> impl IntoView {
        let _ty = ty;
        std::any::type_name::<S>()
    }

    view! {
        GenericComponent::<String> ty={PhantomData};
    };
}
