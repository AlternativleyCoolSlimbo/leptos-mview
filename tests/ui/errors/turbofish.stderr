error: unexpected token `::`
  --> tests/ui/errors/turbofish.rs:14:25
   |
14 |         GenericComponent::<String> ty={PhantomData};
   |                         ^
   |
   = help: turbofish syntax is not used for component generics, place angle brackets directly after the component name
